examples/mechanics/compact_tension_test.cpp

```diff
@@ -0,0 +1,202 @@
+
+#include <fstream>
+#include <iostream>
+
+#include "mpi.h"
+
+#include <Kokkos_Core.hpp>
+
+#include <CabanaPD.hpp>
+
+// Simulate crack propagation in a compact tension test.
+void compactTensionTestExample( const std::string filename )
+{
+    // ====================================================
+    //               Choose Kokkos spaces
+    // ====================================================
+    using exec_space = Kokkos::DefaultExecutionSpace;
+    using memory_space = typename exec_space::memory_space;
+
+    // ====================================================
+    //                   Read inputs
+    // ====================================================
+    CabanaPD::Inputs inputs( filename );
+
+    // ====================================================
+    //                Material parameters
+    // ====================================================
+    double rho0 = inputs["density"];
+    double E = inputs["elastic_modulus"];
+    double nu = 0.25; // Use bond-based model
+    double K = E / ( 3 * ( 1 - 2 * nu ) );
+    double G0 = inputs["fracture_energy"];
+    double delta = inputs["horizon"];
+    delta += 1e-10;
+
+    // ====================================================
+    //                  Discretization
+    // ====================================================
+    std::array<double, 3> low_corner = inputs["low_corner"];
+    std::array<double, 3> high_corner = inputs["high_corner"];
+    std::array<int, 3> num_cells = inputs["num_cells"];
+    int m = std::floor( delta /
+                        ( ( high_corner[0] - low_corner[0] ) / num_cells[0] ) );
+    int halo_width = m + 1; // Just to be safe.
+
+    // ====================================================
+    //                    Pre-notch
+    // ====================================================
+    double height = inputs["system_size"][0];
+    double thickness = inputs["system_size"][2];
+    double L_prenotch = height / 2.0;
+    double y_prenotch = 0.0;
+    Kokkos::Array<double, 3> p0 = { low_corner[0], y_prenotch, low_corner[2] };
+    Kokkos::Array<double, 3> v1 = { L_prenotch, 0, 0 };
+    Kokkos::Array<double, 3> v2 = { 0, 0, thickness };
+    Kokkos::Array<Kokkos::Array<double, 3>, 1> notch_position = { p0 };
+    CabanaPD::Prenotch<1> prenotch( v1, v2, notch_position );
+
+    // ====================================================
+    //                Force model type
+    // ====================================================
+    using model_type = CabanaPD::PMB;
+    using thermal_type = CabanaPD::TemperatureIndependent;
+    using mechanics_type = CabanaPD::ElasticPerfectlyPlastic;
+
+    // ====================================================
+    //    Custom particle generation and initialization
+    // ====================================================
+    // Rectangular prism containing the full specimen: original geometry
```

Comment:
We should never do this - if we need every particle considered, it should use a body term